### PR TITLE
Power manager improvements

### PR DESCRIPTION
**Pull request by szebemate**
This commit does the following
* Do ADC measurement for the battery voltage on power ic 
*  if battery  is present, then enables the battery charging function
* else disable the battery charging
Battery charging status is set BYTE2 bit:0 of CLIENT_DATA
  - 0 means no battery
  - 1 means battery available

### CODE REVIEW

**Comment by effective-range** [comment]
Awesome Mate! Left a couple comments.

File: power_mgr.c [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+
+#define I2C_TRY_CNT 3U
+#define I2C_TMOUT_MS 100U
+typedef enum {
+    kI2C_Dummy,
+    kI2C_Error,
+    kI2C_Success
+}I2CState;
+I2CState i2c_state = kI2C_Dummy;
```
Comment (effective-range) [inline power_mgr.c:35]: This should also be marked as `volatile` I believe

**Review by effective-range — CHANGES_REQUESTED** [review]

File: power_mgr.c [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    //#REG0x30_VBAT_ADC Register bits 1:12
+    tx[0]=0x30;
+    ret += I2CWriteRead(0x6b, tx,1, rx, 2);
+    //value is little endian
+    uint16_t adc_bits = (uint16_t) (rx[0] + (rx[1]<<8));
+    adc_bits &= (0x1ffe);
+    adc_bits = adc_bits >> 1;
+    float adc_mV = adc_bits * 1.99f; //mili Volt
+    
+    //todo what voltage to check
```
Comment (effective-range) [inline power_mgr.c:131]: I think we can calculate that using the Battery Capacity value, and the nominal max  battery voltage which is 4.8 V, 22uF is the battery capacitor and 30 mA is the forced discharge current. Based on that a rough calculation gives ~5 ms, to be on the safe side we can say the we'll wait 10 ms.

**Comment by szebemate** [comment]
Hi! I have fixed the code based on your input. There was also issue with the Charge enable pin which was set in a wrong order. Now CHG_DISA = 0 if charge enabled and CHG_DISA = 1 if charge disabled. Also added delays after enable charge discharge and after enabled ADC sampling. To see it is working please check the push button blinking period. If it blinks in 500msec interval the charging is enabled. If it is blinking with 2000msec period it has disabled the charging.

**Pull request merged**